src/features/account/components/modals/AddToUserCircleModal.tsx

```diff
@@ -26,50 +28,75 @@ export const AddToUserCircleModal: FunctionComponent<
   );
   const dispatch = useDispatch();
 
-  const {
-    isLoading,
-    isVerified,
-    payload,
-    submitTypedSignRequest,
-    signatureError,
-  } = useTypedSignTrustCredential();
+  const [isLoading, setIsLoading] = useState(false);
+  const { issuerAddress, signMessage, signError, accountVCBuilder } =
+    useVerifiableCredential();
 
-  const { showErrorMsg, showSuccessMsg } = useToastMsg();
+  const { showSuccessMsg, showErrorMsg } = useToastMsg();
 
   const shortSubAddress = useMemo(
     () => trimAddress(subjectAddress),
     [subjectAddress],
   );
 
-  useEffect(() => {
-    if (signatureError) {
-      if (signatureError.type === SignatureErrorTypes.Error) {
-        showErrorMsg({
-          title: t`Failed to verify signature`,
-          description: signatureError.message,
-        });
-      } else {
+  const onSignButtonClick = useCallback(() => {
+    if (!issuerAddress) {
+      return;
+    }
+    if (!isAddress(subjectAddress)) {
+      return;
+    }
+    setIsLoading(true);
+
+    const VC = accountVCBuilder.buildAccountTrust(
+      issuerAddress,
+      subjectAddress,
+    );
+
+    signMessage(VC)
+      .then((signature) => {
+        if (signature) {
+          const assertion = accountVCBuilder.getSignedAssertion(VC, signature);
+          console.log('assertion', assertion);
+
+          showSuccessMsg({
+            title: t`Added to your trust circle`,
+            description: t`${shortSubAddress} has been added to your trust circle`,
+          });
+          dispatch(addUserToUserCircle(subjectAddress));
+          dispatch(setAddToUserModalOpen(false));
+        } else {
+          showErrorMsg({
+            title: t`Error`,
+            description: t`The signature is empty`,
+          });
+        }
```

Comment:
when you deny to sign a message, it is not error
```suggestion
      
```

Reply:
will change to `showWarningMsg` here.

Reply:
um? what is the reason so warn the user when he deny to sign?

Reply:
showInfoMsg here now to replace showErrorMsg here.